client/src/components/Form.js

```diff
@@ -0,0 +1,101 @@
+import PropTypes from 'prop-types'
+import InputField from './InputField'
+import Button from './Button'
+import styled from 'styled-components/macro'
+import { useState } from 'react'
+
+Form.propTypes = {
+  onSubmit: PropTypes.func,
+}
+
+export default function Form({ onSubmit }) {
+  const [isDisabled, setIsDisabled] = useState(true)
+
+  return (
+    <StyledForm
+      onSubmit={handleSubmit}
+      onChange={validateForm}
+      aria-label="create a new user"
+      role="form"
+    >
+      <InputField
+        name="name"
+        labelText="name:"
+        placeholder="Please enter your name..."
+        maxLength="40"
+        minLength="5"
+        required
+      />
+      <InputField
+        name="phone"
+        labelText="phone:"
+        placeholder="Please enter your phone number..."
+        minLength="8"
+        required
+      />
+      <InputField
+        name="email"
+        type="email"
+        labelText="email:"
+        placeholder="Please enter your email..."
+        required
+        pattern="^(.+)@(.+)$"
+      />
+      <InputField
+        name="department"
+        labelText="department:"
+        placeholder="Please enter your department..."
+        required
+      />
+      <InputField
+        name="skills"
+        labelText="skills:"
+        placeholder="Please enter your skills..."
+        required
+      />
+
+      <Button disabled={isDisabled} type="submit" children="Create User" />
+    </StyledForm>
+  )
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    const form = event.target
+    const inputs = form.elements
+    const name = inputs.name.value
+    const phone = inputs.phone.value
+    const email = inputs.email.value
+    const department = inputs.department.value
+    const skills = inputs.skills.value.split(', ')
+
+    const newUser = { name, phone, email, department, skills }
+
+    onSubmit(newUser)
+
+    form.reset()
+    event.target.elements.name.focus()
+    setIsDisabled(true)
+  }
+
+  function validateForm(event) {
+    const form = event.target.parentElement.parentElement
+    const inputs = form.elements
+    const inputName = inputs.name.value.trim()
+    const inputPhone = inputs.phone.value.trim()
+    const inputEmail = inputs.email.value.trim()
+    const inputDepartment = inputs.department.value.trim()
+    const inputSkills = inputs.skills.value.trim()
+    setIsDisabled(
+      inputName && inputPhone && inputEmail && inputDepartment && inputSkills
+        ? false
+        : true
+    )
+  }
+}
```

Comment:
Maybe add an error message which would be displayed if the users forgets to fill out a field?

Reply:
All required fields are with a red background. and if they where filled right the background turns green :)